src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -504,6 +506,22 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._tableContainer.classList.remove('hide-username');
 			this._tableContainer.classList.add('hide-password');
 			this._tableContainer.classList.remove('hide-azure-accounts');
+		} else if (currentAuthType === AuthenticationType.dSTSAuth) {
+			this._accountManagementService.getAccountsForProvider('dstsAuth').then(accounts => {
```

Comment:
It'd generally be preferable to convert functions to be async when modifying code like this - then you could replace the .then chains with awaits.

That does change the behavior of when the _token property is populated though so maybe this is intended (see my comment about that below)